modules/3_helpernode/helpernode.tf

```diff
@@ -101,6 +101,55 @@ resource "null_resource" "config" {
             "cd ocp4-helpernode && git checkout ${var.helpernode_tag}"
         ]
     }
+}
+
+resource "null_resource" "prep_helpernode_tools_curl" {
+    triggers = {
+        bootstrap_count = var.bootstrap_port_ip == "" ? 0 : 1
+    }
+    count = length(regexall("\\.zip$", var.helpernode_repo)) > 0 ? 1 : 0
+
+    connection {
+        type        = "ssh"
+        user        = var.rhel_username
+        host        = var.bastion_ip[0]
+        private_key = var.private_key
+        agent       = var.ssh_agent
+        timeout     = "${var.connection_timeout}m"
+        bastion_host = var.jump_host
+    }
+
+    provisioner "remote-exec" {
+        inline = [
+            "mkdir -p .openshift",
+            "rm -rf ocp4-helpernode",
+            "echo 'Downloading ocp4-helpernode...'",
+            "curl -o ocp4-helpernode.zip ${var.helpernode_repo}",
+            "echo 'Extracting ocp4-helpernode...'",
+            "unzip ocp4-helpernode.zip",
+            "mv ocp4-helpernode-master ocp4-helpernode",
```

Comment:
It won't be `ocp4-helpernode-master` always. May be we can use a wildcard?

Reply:
I tried that before at a customer and we had some trouble with it, because it can happen that source and target folder are equal. But this morning I had the idea of using a helper folder:
            "mkdir -p .openshift",
            "rm -rf ocp4-helpernode",
            "rm -rf ocp4-extract-helper",
            "mkdir -p ocp4-extract-helper", 
            "echo 'Downloading ocp4-helpernode...'",
            "curl -o ocp4-extract-helper/ocp4-helpernode.zip ${var.helpernode_repo}",
            "echo 'Extracting ocp4-helpernode...'",
            "cd ocp4-extract-helper && unzip ocp4-helpernode.zip",
            "rm -rf ocp4-extract-helper/ocp4-helpernode.zip",
            "mv ocp4-extract-helper/ocp4-helpernode* ocp4-helpernode",
            "rm -rf ocp4-extract-helper"
Might look a little bit strange, but as long as it solves the issue...